pennylane_lightning/src/tests/Test_Measures.cpp

```diff
@@ -0,0 +1,217 @@
+#include <complex>
+#include <stdio.h>
+#include <vector>
+
+#include "Measures.hpp"
+#include "StateVector.hpp"
+#include "Util.hpp"
+
+#include <catch2/catch.hpp>
+
+using namespace Pennylane;
+StateVector<double> Initializing_StateVector() {
+    // Defining a StateVector in a non-trivial configuration:
+    size_t num_qubits = 3;
+    size_t data_size = std::pow(2, num_qubits);
+
+    std::complex<double> *arr;
+    arr = new complex<double>[data_size]();
+    arr[0] = 1;
```

Comment:
Use `StateVectorManaged` where possible, or ensure memory is allocated by `std::vector` so that it is freed upon destruction.

Reply:
Nice, I will give a look into the `StateVectorManaged`, to better understand it use.